apps/gdalalg_raster_rgb_to_palette.cpp

```diff
@@ -0,0 +1,323 @@
+/******************************************************************************
+ *
+ * Project:  GDAL
+ * Purpose:  gdal "raster rgb-to-palette" subcommand
+ * Author:   Even Rouault <even dot rouault at spatialys.com>
+ *
+ ******************************************************************************
+ * Copyright (c) 2025, Even Rouault <even dot rouault at spatialys.com>
+ *
+ * SPDX-License-Identifier: MIT
+ ****************************************************************************/
+
+#include "gdalalg_raster_rgb_to_palette.h"
+
+#include "cpl_string.h"
+#include "gdal_alg.h"
+#include "gdal_priv.h"
+#include "commonutils.h"
+
+#include <array>
+
+//! @cond Doxygen_Suppress
+
+#ifndef _
+#define _(x) (x)
+#endif
+
+/************************************************************************/
+/*                    GDALRasterRGBToPaletteAlgorithm()                 */
+/************************************************************************/
+
+GDALRasterRGBToPaletteAlgorithm::GDALRasterRGBToPaletteAlgorithm()
+    : GDALAlgorithm(NAME, DESCRIPTION, HELP_URL)
+{
+    AddProgressArg();
+
+    AddOpenOptionsArg(&m_openOptions);
+    AddInputFormatsArg(&m_inputFormats)
+        .AddMetadataItem(GAAMDI_REQUIRED_CAPABILITIES, {GDAL_DCAP_RASTER});
+    AddInputDatasetArg(&m_inputDataset, GDAL_OF_RASTER);
+
+    AddOutputDatasetArg(&m_outputDataset, GDAL_OF_RASTER);
+    AddOutputFormatArg(&m_format, /* bStreamAllowed = */ false,
+                       /* bGDALGAllowed = */ false)
+        .AddMetadataItem(GAAMDI_REQUIRED_CAPABILITIES,
+                         {GDAL_DCAP_RASTER, GDAL_DCAP_CREATE});
+    AddCreationOptionsArg(&m_creationOptions);
+    AddOverwriteArg(&m_overwrite);
+
+    AddArg("color-count", 0,
+           _("Select the number of colors in the generated color table"),
+           &m_colorCount)
+        .SetDefault(m_colorCount)
+        .SetMinValueIncluded(2)
+        .SetMaxValueIncluded(256);
+    AddArg("color-map", 0, _("Color map filename"), &m_colorMap);
+}
+
+/************************************************************************/
+/*                GDALRasterRGBToPaletteAlgorithm::RunImpl()            */
+/************************************************************************/
+
+bool GDALRasterRGBToPaletteAlgorithm::RunImpl(GDALProgressFunc pfnProgress,
+                                              void *pProgressData)
+{
+    auto poSrcDS = m_inputDataset.GetDatasetRef();
+    CPLAssert(poSrcDS);
+
+    if (m_format.empty())
+    {
+        m_format = GetOutputDriverForRaster(m_outputDataset.GetName().c_str());
+        if (m_format.empty())
+        {
+            ReportError(CE_Failure, CPLE_AppDefined,
+                        "Cannot guess output driver from output filename");
+            return false;
+        }
+    }
+    if (EQUAL(m_format.c_str(), "JPEG") || EQUAL(m_format.c_str(), "WEBP"))
```

Comment:
Don't we have a capability to query for color table support? 
This approach seems a bit fragile to me.

Reply:
> Don't we have a capability to query for color table support?

no, but (most )drivers that do not support color table will error out. Here I anticipated for quite common use cases to emit a clean error message